src/inmanta/data/__init__.py

```diff
@@ -4855,44 +4855,37 @@ async def get_resource_states_latest_version(
 
     @stable_api
     @classmethod
-    async def get_current_resource_state(cls, env: uuid.UUID, rid: ResourceIdStr) -> Optional[ResourceState]:
+    async def get_current_resource_state(cls, env: uuid.UUID, rid: ResourceIdStr) -> Optional[const.ResourceState]:
         """
         Return the state of the given resource in the latest version of the configuration model
         or None if the resource is not present in the latest version.
         """
         query = """
-            WITH latest_released_version AS (
-                SELECT max(version) AS version
-                FROM configurationmodel
-                WHERE environment=$1 AND released
-            )
-            SELECT (
+            SELECT  (
                 CASE
-                    -- The resource_persistent_state.last_non_deploying_status column is only populated for
-                    -- actual deployment operations to prevent locking issues. This case-statement calculates
-                    -- the correct state from the combination of the resource table and the
-                    -- resource_persistent_state table.
-                    WHEN r.status::text IN('deploying', 'undefined', 'skipped_for_undefined')
-                        -- The deploying, undefined and skipped_for_undefined states are not tracked in the
-                        -- resource_persistent_state table.
-                        THEN r.status::text
-                    WHEN rps.last_deployed_attribute_hash != r.attribute_hash
-                        -- The hash changed since the last deploy -> new desired state
-                        THEN r.status::text
-                        -- No override required, use last known state from actual deployment
-                        ELSE rps.last_non_deploying_status::text
+                    WHEN rps.is_orphan = TRUE
+                        THEN 'orphaned'
```

Comment:
I *think* it would be more direct and efficient to replace this with a `AND NOT rps.is_orphan` in the `WHERE` clause.
Efficiency is probably not too much of a concern here, but still, good practice.